btb_phylo.py

```diff
@@ -339,24 +340,56 @@ def build_tree(tree_path, snp_sites_outpath):
+    parser.add_argument("--config", type=str, default=None)
+    parser.add_argument("--sample_name", "-s", dest="Sample", type=str, nargs="+")
+    parser.add_argument("--clade", "-c", dest="group", type=str, nargs="+")
+    parser.add_argument("--pcmapped", "-pc", dest="pcMapped", type=float, nargs=2)
+    parser.add_argument("--genomecov", "-gc", dest="GenomeCov", type=float, nargs=2)
+    parser.add_argument("--n_count", "-nc", dest="Ncount", type=float, nargs=2)
     parser.add_argument("--flag", "-f", dest="flag", type=str, nargs="+")
-    kwargs = vars(parser.parse_args())
-    warnings.formatwarning = utils.format_warning
-    multi_fasta_path = "/home/nickpestell/tmp/test_multi_fasta.fas"
-    results_path = "/home/nickpestell/tmp/"
-    df_summary = get_samples_df("s3-staging-area", "nickpestell/btb_wgs_samples.csv", **kwargs)
-    # save df_summary (samples to include in VB) to csv
-    df_summary.to_csv(os.path.join(results_path, "summary.csv"))
-    # TODO: make multi_fasta_path a tempfile and pass file object into build_multi_fasta
+    parser.add_argument("--meandepth", "-md", dest="MeanDepth", type=float, nargs=2)
+    # parse agrs
+    clargs = vars(parser.parse_args())
+    # retreive "non-filtering" args
+    results_path = clargs.pop("results_path")
+    tree = clargs.pop("build_tree")
+    config = clargs.pop("config")
+    # if config json file provided
+    if config:
+        error_keys = [key for key, val in clargs.items() if val]
+        # if any arguments provided with --config
+        if any(error_keys):
+            raise parser.error(f"arguments '{', '.join(error_keys)}' are incompatible with "
+                               "the 'config' argument")
+        # parse config file
+        with open(config) as f:
+            kwargs = json.load(f)
+    else:
+        # remove unused filtering args
+        kwargs = {k: v for k, v in clargs.items() if v is not None}
+    # set output paths
+    summary_csv_path = os.path.join(results_path, "sample_summary.csv")
+    multi_fasta_path = os.path.join(results_path, "multi_fasta.fas")
     snp_sites_outpath = os.path.join(results_path, "snps.fas")
     snp_dists_outpath = os.path.join(results_path, "snp_matrix.tab")
     tree_path = os.path.join(results_path, "mega")
-    build_multi_fasta(multi_fasta_path, df_summary)
+    # get samples from btb_wgs_samples.csv and filter
+    samples_df = get_samples_df("s3-staging-area", "nickpestell/btb_wgs_samples.csv", **kwargs)
+    # save df_summary (samples to include in VB) to csv
+    samples_df.to_csv(summary_csv_path)
+    # concatonate fasta files
+    build_multi_fasta(multi_fasta_path, samples_df)
+    # run snp-sites
     snp_sites(snp_sites_outpath, multi_fasta_path)
-    build_snp_matrix(snp_dists_outpath, snp_sites_outpath, threads=4)
-    #build_tree(tree_path, snp_sites_outpath)
+    # run snp-dists
+    build_snp_matrix(snp_dists_outpath, snp_sites_outpath)
```

Comment:
Not using 4 core? or is that specified in build_snp_matrix

Reply:
Good spot, I was actually planning on putting an optional argument in for this, but forgot!

Reply:
Will add now